### PR TITLE
UTF-8 support

### PR DESCRIPTION
**Pull request by bonfante**
UTF-8 support.
----
Extension mbstring become required and php ^7.2 to use mb_ord

### CODE REVIEW

File: src/Hashids.php [hunk 1]
```diff
@@ -91,54 +92,52 @@ class Hashids implements HashidsInterface
      * Create a new hashids instance.
      *
      * @param string $salt
-     * @param int $minHashLength
+     * @param int    $minHashLength
```
Comment (vinkla) [inline src/Hashids.php]: Please remove the tab.

File: src/Hashids.php [hunk 2]
```diff
@@ -403,18 +400,30 @@ protected function unhash($input, $alphabet)
...
     }
+
+    /**
+     * replace simple use of $this->multiByteSplit with multi byte string.
+     *
+     * @param $string
+     *
+     * @return string[]|array
+     */
+    private function multiByteSplit($string): array
```
Comment (vinkla) [inline src/Hashids.php]: Please change this to protected.

File: src/Hashids.php [hunk 3]
```diff
@@ -403,18 +400,30 @@ protected function unhash($input, $alphabet)
...
-        if (extension_loaded('bcmath')) {
+        if (\extension_loaded('bcmath')) {
             return new Bc();
         }
 
         throw new RuntimeException('Missing BC Math or GMP extension.');
     }
+
+    /**
+     * replace simple use of $this->multiByteSplit with multi byte string.
```
Comment (vinkla) [inline src/Hashids.php]: Please use uppercase first letter.

File: composer.json [hunk 4]
```diff
@@ -27,7 +27,8 @@
         }
     ],
     "require": {
-        "php": "^7.1.3"
+        "php": "^7.2.0",
```
Comment (vinkla) [inline composer.json]: Please change to `^7.2`.

**Comment by vinkla** [comment]
Thanks for the pull request @bonfante!

This pull request is featuring breaking changes and will require a new major release.

@ivanakimov any thoughts?

**Comment by ivanakimov** [comment]
@bonfante Thanks for the PR!
@vinkla Thanks for the quick follow-up -- I am following your lead on this one!

**Comment by bonfante** [comment]
@vinkla done 👍.

**Comment by vinkla** [comment]
Thanks @bonfante! ✨ 

We'll bump the branch alias to 4.0, please try using the new version in any project. We'll release a new version in a week or two.

**Comment by jfcherng** [comment]
`mb_ord()` could be polyfill thus no need for PHP72-only actually.

https://github.com/symfony/polyfill-php72/blob/master/Php72.php#L191

**Comment by vinkla** [comment]
That is great! We’ll probably leave it as it is since 7.1 will reach EOL this year anyway.

**Pull request merged**